autopilot/scanner/scanner.go

```diff
@@ -127,14 +146,20 @@ func (s *scanner) Scan(ctx context.Context, hs HostScanner, force bool) {
 }
 
 func (s *scanner) Shutdown(ctx context.Context) error {
+	s.mu.Lock()
+	if !s.scanning {
+		s.mu.Unlock()
+		return nil
+	}
+	s.scanningCtxCancel(ErrShuttingDown)
```

Comment:
If you get rid of `scanning`, closing the context is all we need here. Then we'd block on the wait group until any remaining goroutines exit and we are good. No need to exit early.

Reply:
The `scanning` field didn't really bother me all that much. It's a field you'd expect and we need it for returning the `Status`. It's also quite useful in the sense that I don't have to check whether `scanningCtxCancel` is `nil` and I don't have to initialise the field with a dummy value to ensure it's always safe to call it.